src/KafkaFlow.Retry/Durable/Polling/Jobs/CleanupPollingJob.cs

```diff
@@ -0,0 +1,74 @@
+﻿namespace KafkaFlow.Retry.Durable.Polling.Jobs
+{
+    using System;
+    using System.Threading.Tasks;
+    using Dawn;
+    using KafkaFlow.Retry.Durable.Definitions.Polling;
+    using KafkaFlow.Retry.Durable.Repository;
+    using KafkaFlow.Retry.Durable.Repository.Actions.Delete;
+    using KafkaFlow.Retry.Durable.Repository.Model;
+    using Quartz;
+
+    [DisallowConcurrentExecutionAttribute]
+    internal class CleanupPollingJob : IJob
+    {
+        public async Task Execute(IJobExecutionContext context)
+        {
+            var jobDataMap = context.JobDetail.JobDataMap;
+
+            Guard.Argument(jobDataMap.ContainsKey(PollingJobConstants.RetryDurableQueueRepository), PollingJobConstants.RetryDurableQueueRepository)
+                .True("Argument RetryDurableQueueRepository wasn't found and is required for this job");
+
+            Guard.Argument(jobDataMap.ContainsKey(PollingJobConstants.CleanupPollingDefinition), PollingJobConstants.CleanupPollingDefinition)
+                .True("Argument CleanupPollingDefinition wasn't found and is required for this job");
+
+            Guard.Argument(jobDataMap.ContainsKey(PollingJobConstants.LogHandler), PollingJobConstants.LogHandler)
+                .True("Argument LogHandler wasn't found and is required for this job");
+
+            var retryDurableQueueRepository = jobDataMap[PollingJobConstants.RetryDurableQueueRepository] as IRetryDurableQueueRepository;
+            var cleanupPollingDefinition = jobDataMap[PollingJobConstants.CleanupPollingDefinition] as CleanupPollingDefinition;
+            var logHandler = jobDataMap[PollingJobConstants.LogHandler] as ILogHandler;
+            var schedulerId = jobDataMap[PollingJobConstants.SchedulerId] as string;
+
+            Guard.Argument(retryDurableQueueRepository).NotNull();
+            Guard.Argument(cleanupPollingDefinition).NotNull();
+            Guard.Argument(logHandler).NotNull();
+            Guard.Argument(schedulerId).NotNull().NotEmpty();
```

Comment:
You are validating twice the existence of the parameters. Let me suggest the following:

- Remove the first guards (and move the messages to the guards bellow perhaps?)
- Replace the dictionary get by index with the `TryGetValue()` method.
- Use the guards bellow to validate the parameters.

What do you think?

Reply:
Please look at the new code, which uses a new extension for this.